src/readers/morphologyHDF5.cpp

```diff
@@ -140,58 +98,73 @@ void MorphologyHDF5::_resolveV1() {
     _sections.reset(new HighFive::DataSet(_group.getDataSet(_d_structure)));
 }
 
-bool MorphologyHDF5::_readV11Metadata() {
-    uint32_t family;
-
+/* Heuristic to determine the version
+ - Look for the version field (which exists only for h5v1.1 and h5v1.2)
+ - Try to resolve the h5v1 point dataset, if it works and no version field was previously found
+   then it must be h5v1.0
+   Returns in case of successful resolution
+ - If the resolution fails, tries to find the h5v2 version field (which is located elsewhere)
+   throw a custom message if h5v2 is detected
+ - in any other case, throws a generic message
+*/
+void MorphologyHDF5::_readMetadata(const std::string& source) {
+    // h5v1.1 and h5v2 attempt
     try {
         HighFive::SilenceHDF5 silence;
         const auto metadata = _group.getGroup(_g_metadata);
         const auto attr = metadata.getAttribute(_a_version);
 
         uint32_t version[2];
         attr.read(version);
-        if (version[0] != 1 || version[1] != 1)
-            return false;
-
-        const auto familyAttr = metadata.getAttribute(_a_family);
-        familyAttr.read(family);
-    } catch (const HighFive::GroupException&) {
-        return false;
-    } catch (const HighFive::Exception& e) {
-        // All other exceptions are not expected because if the metadata
-        // group exits it must contain at least the version, and for
-        // version 1.1 it must contain the family.
-        throw morphio::RawDataError("Error reading morphology " + _uri + "  metadata." + e.what());
+        if (version[0] == 1 && version[1] >= 1) {
```

Comment:
Shouldn't we add an upper limit for the minor version number ?
Only 1.1 and 1.2 are supported and we want to throw if a 1.3 arrives one day, no ?

Reply:
Yes, I was wondering actually. Indeed, it is probably safer to raise.